### PR TITLE
Add TimeZone spec to `Backup` and `SqlJob` resources

### PR DESCRIPTION
**Pull request by mbezhanov**
Closes https://github.com/mariadb-operator/mariadb-operator/issues/383.

This PR implements the request from #383.

It also introduces a couple of changes that we agreed upon in previous PRs:

1. Addition of a `CronJobTemplate` type to encapsulate the `TimeZone`, `SuccessfulJobsHistoryLimit`, and `FailedJobsHistoryLimit` fields.
2. Heavy refactoring of `backup_controller_test.go` and `sqljob_controller_test.go` to cover additional test cases and reduce code duplication.

I was also thinking about refactoring `SqlJob` to use `BatchReconciler` in this PR, but the PR became kind of large, so I'll leave it for some other time if that's okay.

A few things to point out:

- I haven't run the tests on anything other than Kubernetes 1.30 yet, but plan on doing so in the following days.
- I intentionally left the individual commits to (hopefully) make the review a bit easier, but I'm ready to squash them before merging.
- I can further split the refactoring changes into a separate PR if preferred.

I remain available if any additional changes are necessary.

### CODE REVIEW

**Comment by mbezhanov** [comment]
By the way, I had to make some manual tweaks to the KIND cluster to get the integration tests running locally (had to manually provision a MinIO access key and then create the respective Kubernetes secret for use with the S3 bucket tests).

Is there a procedure that provisions this automatically?

**Comment by mbezhanov** [comment]
Tested on all versions between (and including) `1.23` and `1.30`. `TimeZone` field is silently dropped on `1.23` and `1.24` and successfully set on the rest as expected.

The refactored integration tests run successfully on everything >= `1.25`. They fail on <= `1.24`, but only on the `TimeZone` assertions, as the field is ignored, so they are never met (this is expected).

Everything looks good on my end. The PR is ready for review.

File: internal/controller/backup_controller_test.go [hunk 1]
```diff
@@ -18,81 +20,135 @@ var _ = Describe("Backup", func() {
...
+		),
+		Entry("should reconcile a CronJob with PVC storage and history limits",
+			"backup-pvc-scheduled-with-limits-test",
+			applyDecoratorChain[*mariadbv1alpha1.Backup](
+				getBackupWithPVCStorage,
+				decorateBackupWithSchedule,
+				decorateBackupWithHistoryLimits,
+			),
+			testBackup,
+		),
```
Comment (mmontes11) [inline internal/controller/backup_controller_test.go:85]: This is brilliant! ✨

File: internal/controller/backup_controller_test.go [hunk 2]
```diff
@@ -18,81 +20,135 @@ var _ = Describe("Backup", func() {
...
-		key := types.NamespacedName{
-			Name:      "backup-s3-scheduled-test-prefix",
-			Namespace: testNamespace,
-		}
-		backup := getBackupWithScheduleAndHistoryLimits(
-			getBackupWithS3Storage(key, "test-backup", "mariadb"),
-		)
-		testS3Backup(backup)
-	})
+	DescribeTable("Creating a Backup",
```
Comment (mmontes11) [inline internal/controller/backup_controller_test.go:23]: Much clear and readable 🌟

**Review by mmontes11 — APPROVED** [review]
Thank you very much for this brilliant contribution! ✨ 🌟 

LGTM! 🚀 

The refactor is massively helpful, not only because it provides re-usability and more clarity, but also because we will be having more `CronJob` objects in the future to implement scheduled physical backups for [PITR](https://github.com/mariadb-operator/mariadb-operator/issues/507)! Thank you very much for taking the time, it is very much appreciated!

I will be documenting this together with the [`MariaDB` timeZone support](https://github.com/mariadb-operator/mariadb-operator/pull/726) and release it in `v0.0.30`. Stay tuned! 

> The refactored integration tests run successfully on everything >= 1.25. They fail on <= 1.24, but only on the TimeZone assertions, as the field is ignored, so they are never met (this is expected).

Thank you for the effort in testing with multiple Kubernetes versions! At some point, we will implement end-to-end (e2e) matrix testing to be executed before the release to ensure we don't miss any version-related regressions. We will focus on non-EOL versions of Kubernetes, as listed here:
- https://endoflife.date/kubernetes
Therefore, the timeZone tests not working on versions <= 1.24 shouldn't be a concern, as these versions are quite old and have been EOL for a while.

**Pull request merged**